pkg/azuredx/client/client.go

```diff
@@ -0,0 +1,120 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/google/uuid"
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/grafana/azure-data-explorer-datasource/pkg/azuredx/models"
+	"github.com/grafana/azure-data-explorer-datasource/pkg/azuredx/tokenprovider"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	sdkhttpclient "github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/hashicorp/go-hclog"
+)
+
+// Client is an http.Client used for API requests.
+type Client struct {
+	*http.Client
+	Log hclog.Logger
+}
+
+// NewClient creates a Grafana Plugin SDK Go Http Client
+func New(tokenProvider *tokenprovider.AccessTokenProvider) (*Client, error) {
+	c := Client{}
+	var err error
+
+	c.Client, err = sdkhttpclient.New(sdkhttpclient.Options{
+		Middlewares: []sdkhttpclient.Middleware{
```

Comment:
This will give me all default values from the SDK Http Client, right? Should be no need for me to set time out etc?